msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java

```diff
@@ -2269,15 +2270,15 @@ public void processPvtMsgReplySentAction(PrivateMessage currentMessage, PrivateM
   private String encrypt(String value) {
 	  try {
 		  IvParameterSpec iv = new IvParameterSpec(INIT_VECTOR.getBytes(StandardCharsets.UTF_8));
-		  String key = serverConfigurationService.getString("msgcntr.no.reply.secret", "1111111111111111");
-		  SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes(StandardCharsets.UTF_8), "AES");
+		  String key = serverConfigurationService.getString("msgcntr.no.reply.secret", "11111111");
+		  SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes(StandardCharsets.UTF_8), "DES");
```

Comment:
Why are we switching to DES here?

Reply:
let's use serverName as the default instead of 111111

Reply:
The RFC specification for email addresses states that the local part of the address cannot exceed 64 characters.  For this reason, if we use an AES algorithm, because it requires at least 16 characters for its key and its seed, the encryption that returns when it is converted to hexadecimal and the mail identification header is added, it can sometimes exceed those 64 characters.
For this reason we have changed the encryption algorithm as DES uses 8 characters for the key and the seed, generating more manageable values for the local part of the email address.
Regarding the key, it must have 8 characters.  It can not be used serverName because it will have a variable length depending on the client that uses it.  When enabling this functionality, the necessary property must be enabled and, therefore, this "msgcntr.no.reply.secret" can also be enabled in default.sakai.properties.  However, this value "11111111" is defined by default in case you have not enabled the property "msgcntr.no.reply.secret".

Reply:
I think we cannot use the serverName by default because in cluster environments this value will be different for each instance so you will have to decrypt the key in the same instance you use for create it.
On the other hand you cannot use AES and remove characters to fit to 64 bytes because later you won't be able to decrypt the key.

Reply:
serverName should be identical across all nodes in a cluster.

I think using one central secret is fine, e.g., "sakai.encryption.secret" and then we can use it in other places like BasicLTI

Reply:
🤔 Sorry I was thinking in **serverId**, anyway using a common secret make sense to me, but you still have to use a "default" value for an OOTB install. Maybe using `serverConfigurationService.getString("sakai.encryption.secret")` and adding a default value '111111' in [kernel.properties](https://github.com/sakaiproject/sakai/blob/master/kernel/component-manager/src/main/bundle/org/sakaiproject/config/kernel.properties) is a better solution?

Reply:
@ottenhoff  .... could you take a look on this?

Reply:
This is what makes most sense to me:

serverConfigurationService.getString("sakai.encryption.secret", serverConfigurationService.getServerName())

Reply:
Thanks, @ottenhoff 👍 :) ... I'm going to talk to @victorGomollon